network/onchain-discovery/src/test.rs

```diff
@@ -354,15 +353,23 @@ fn queries_storage_on_tick() {
     let update_reqs = update_reqs
         .into_iter()
         .map(|req| match req {
-            ConnectivityRequest::UpdateAddresses(_src, peer_id, addrs) => (peer_id, addrs),
+            ConnectivityRequest::UpdateAddresses(_src, updated_addresses) => updated_addresses,
             _ => panic!(
                 "Unexpected ConnectivityRequest, expected UpdateAddresses: {:?}",
                 req
             ),
         })
-        .collect::<HashMap<_, _>>();
+        .collect::<Vec<_>>();
 
-    assert_eq!(expected_update_reqs, update_reqs);
+    // The first UpdateAddress request should update all addresses.
+    if let Some(updated_addresses) = update_reqs.first() {
+        assert_eq!(expected_update_reqs, *updated_addresses);
+    }
+    // There should be exactly one UpdateAddress request.
+    match expected_update_reqs.len() {
+        x if x > 0 => assert_eq!(1, update_reqs.len()),
+        _ => assert_eq!(0, update_reqs.len()),
+    }
```

Comment:
let's simplify this a bit!

```rust
let expected_conn_mgr_reqs = vec![
    ConnectivityRequest::UpdateAddresses(DiscoverySource::OnChain, expected_update_reqs),
];

let conn_mgr_reqs = rt.block_on(conn_mgr_reqs_rx.collect::<Vec<_>>());
assert_eq!(expected_conn_mgr_reqs, conn_mgr_reqs);
```

Reply:
ahh darn, nevermind then. :(

removing `GetDialQueueSize` would be a good target for refactoring. we might need stricter separation between the internal ( input messages -> state machine -> output messages ) process and the actor's event loop which feeds into the state machine. specifically, we might need a stronger mapping b/w input -> output messages. we probably also need to mock the delay timers in the `pending_dials` queue somehow.

some potential inspiration: https://sled.rs/simulation.html